File: Software Technologies/Calculators/C#/CalculatorApp/Models/Calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CalculatorApp.Models
{
    public class Calculator
    {
        public Calculator()
        {
            this.Result = 0;
        }
        public decimal LeftOperand { set; get; }
        public decimal RightOperand { set; get; }
        public string Operator{ set; get; }
        public decimal Result { set; get; }
        public void CalculateResult()
        {
            switch(this.Operator)
            {
                case "+":
                    this.Result = this.LeftOperand + this.RightOperand;
                    break;
                case "-":
                    this.Result = this.LeftOperand - this.RightOperand;
                    break;
                case "/":
                    this.Result = this.LeftOperand / this.RightOperand;
                    break;
                case "*":
                    this.Result = this.LeftOperand * this.RightOperand;
                    break;
            }
        }
    }
    
}

File: Software Technologies/Calculators/JAVA/src/main/java/com/softuni/entity/Calculator.java
package com.softuni.entity;

public class Calculator{
    public Calculator(double leftOperand,double rightOperand, String operator)
    {
        this.leftOperand=leftOperand;
        this.rightOperand=rightOperand;
        this.operator=operator;
    }
    public double calculateResult()
    {
        double result=0;
        switch(this.operator)
        {
            case "+":
                result=this.getLeftOperand()+this.getRightOperand();
                break;
            case "-":
                result=this.getLeftOperand()-this.getRightOperand();
                break;
            case "*":
                result=this.getLeftOperand()*this.getRightOperand();
                break;
            case "/":
                result=this.getLeftOperand()/this.getRightOperand();
                break;
            case "%":
                result=this.getLeftOperand()%this.getRightOperand();
                break;
            case "power":
                result=Math.pow(this.getLeftOperand(),this.getRightOperand());
                break;

        }
        return result;
    }
    private double leftOperand;
    private String operator;

    public double getLeftOperand() {
        return leftOperand;
    }

    public void setLeftOperand(double leftOperand) {
        this.leftOperand = leftOperand;
    }

    public String getOperator() {
        return operator;
    }

    public void setOperator(String operator) {
        this.operator = operator;
    }

    public double getRightOperand() {
        return rightOperand;
    }

    public void setRightOperand(double rightOperand) {
        this.rightOperand = rightOperand;
    }

    private double rightOperand;


}
